lib/rucio/db/sqla/session.py

```diff
@@ -303,21 +306,26 @@ def new_funct(*args, **kwargs):
             raise RucioException('read_session decorator should not be used with generator. Use stream_session instead.')
 
         if not kwargs.get('session'):
-            session = get_session()
-            try:
-                kwargs['session'] = session
-                return function(*args, **kwargs)
-            except TimeoutError as error:
-                session.rollback()  # pylint: disable=maybe-no-member
-                raise DatabaseException(str(error))
-            except DatabaseError as error:
-                session.rollback()  # pylint: disable=maybe-no-member
-                raise DatabaseException(str(error))
-            except:
-                session.rollback()  # pylint: disable=maybe-no-member
-                raise
-            finally:
-                session.remove()
+            # using sqlAlchemy 2.0 format
+            # the scoped_session vs context manager problem.
+            # see https://github.com/sqlalchemy/sqlalchemy/issues/6519
```

Comment:
I don't understand how this github thread is relevant and why it is linked in multiple places of this pull request. I personally don't see any problem at all. Just somebody trying to incorrectly use the sqlalchemy objects. Sqlalchemy can be very confusing with all the Session vs SessionMaker vs scoped_session thing, but I wouldn't put this github link here. It has potential to make people think that there is some bug/issue in sqlalchemy for which we implemented a workaround here. And it's not the case

Reply:
It is a "problem" in the sense of using a context manager for the `scoped session`, but the problem cannot be fixed in SQLalchemy. Because of this, one cannot write code as below
```
with Session.begin() as session:
    session.add(Test(value="foo"))
```
instead,  one has to do 
```
with Session() as session:
     with session.begin():
         session.add(Test(value="foo"))
```
It took me some time to find out this, so I thought it will help people when they want to be more succinct.  I may just leave one link instead of in all py files.

Reply:
The link removed.